lint_test/proper_super_calls_test.dart

```diff
@@ -0,0 +1,74 @@
+// ignore_for_file: prefer_match_file_name, no_empty_block
+
+class Widget {}
+
+abstract class State<T> {
+  build();
+// ignore: proper_super_calls
+  dispose() {}
+// ignore: proper_super_calls
+  initState() {}
+}
```

Comment:
This highlights a problem.

It's quite common to have a `dispose` method on custom classes that have some resources to release.

They don't necessarily override anything. We should not lint these classes.